frontend/dockerfile/linter/generate.go

```diff
@@ -0,0 +1,133 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+type Rule struct {
+	Name        string
+	Description string
+}
+
+const tmplStr = `---
+title: {{.Rule.Name}}
+description: {{.Rule.Description}}
+---
+
+{{.Content}}
```

Comment:
here's a far out ask. Is there any way we can generate the sample error using the `Format` field? https://github.com/moby/buildkit/blob/master/frontend/dockerfile/linter/linter.go#L79

Just thinking if there's a way to make sure the sample error has the same prose as what's defined on the struct of the error. If generating the sample error isn't easily doable, maybe we could update the tests to check that generated errors have the same format as what's in the doc.

Reply:
I think we could but where to put `Format` in the template and what layout would this have?

Reply:
I wonder if we could put the templating logic in the Markdown files directly. For example, consider `frontend/dockerfile/linter/docs/RuleName.md`

```
---
title: {{ .Name }}
description: {{ .Description }}
---

Example warning:

{{ .Format "foo" "bar" }}

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod...
```

Reply:
Yes sounds good